src/components/Books.js

```diff
@@ -1,24 +1,53 @@
-import React, { useRef } from 'react';
-import { useDispatch } from 'react-redux';
-import uuid from 'react-uuid';
-import { addBook, removeBook } from '../redux/books/books';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook, removeBook, allBooks } from '../redux/books/books';
 
 function Books() {
   const dispatch = useDispatch();
-  const id = uuid();
-  const title = useRef(null);
-  const author = useRef(null);
+  const books = useSelector(allBooks);
+  const [title, addTitle] = useState(' ');
+  const [author, addAuthor] = useState(' ');
 
-  const submitBookToStore = () => {
-    dispatch(addBook(title.current.value, author.current.value, id));
+  const addBookTitle = (e) => {
+    addTitle(e.target.value);
+  };
+
+  const addBookAuthor = (e) => {
+    addAuthor(e.target.value);
+  };
+
+  const submitBookToStore = (e) => {
+    e.preventDefault();
+    dispatch(addBook({ title, author, id: uuidv4() }));
+    addTitle('');
+    addAuthor('');
+  };
+
+  const removeBookFromStore = (e) => {
+    dispatch(removeBook({ id: e.target.id }));
   };
 
   return (
     <div>
-      <input ref={title} type="text" placeholder="Title" />
-      <input ref={author} type="text" placeholder="Author " />
-      <button type="submit" onClick={submitBookToStore}>Add Book</button>
-      <button type="submit" onClick={() => dispatch(removeBook(id))}>Remove Book</button>
+      <h4>ADD A NEW BOOK</h4>
+      <div>
+        <ul>
+          { books && books.map((book) => (
+            <li key={book.id}>
+              <span>{book.title}</span>
+              <button id={book.id} type="button" onClick={removeBookFromStore}>Delete</button>
+            </li>
+          ))}
```

Comment:
- [ ] Good job implementing each book! Now, let's change it a little bit. In place to only HTML elements for each book, why not create a react component for each book? Here you can call it `<Book />` and then add all the info with more attributes. 🐶

Reply:
Created the book component